bifrost_min_read_check/datadump.py

```diff
@@ -1,22 +1,44 @@
-from bifrostlib import datahandling
+from bifrostlib import common
+from bifrostlib.datahandling import Sample
+from bifrostlib.datahandling import SampleComponentReference
+from bifrostlib.datahandling import SampleComponent
+from bifrostlib.datahandling import Category
+from typing import Dict
+import os
 
 
-def extract_has_min_num_of_reads(sampleComponentObj):
-    import re
-    summary, results, file_path, key = sampleComponentObj.start_data_extraction("has_min_num_of_reads")
-    buffer = datahandling.read_buffer(file_path)
-    results[key]["has_min_num_of_reads"] = "has_min_num_of_reads:True" in buffer
-    results[key]["min_read_num"] = int(re.search("min_read_num:\s*([0-9]+)", buffer, re.MULTILINE).group(1))
-    summary["has_min_num_of_reads"] = results[key]["has_min_num_of_reads"]
-    return (summary, results)
+def extract_has_min_num_of_reads(size_check: Category, results: Dict, component_name: str) -> None:
+    file_name = "has_min_num_of_reads"
+    file_key = common.json_key_cleaner(file_name)
+    file_path = os.path.join(component_name, file_name)
+    results[file_key] = {}
+    results[file_key]["has_min_num_of_reads"] = "True" in common.get_group_from_file("has_min_num_of_reads:(True|False)", file_path)
+    results[file_key]["num_of_reads"] = common.get_group_from_file("min_read_num:\s*([0-9]+)", file_path)
+    size_check["summary"]["has_min_num_of_reads"] = results[file_key]["has_min_num_of_reads"]
+    size_check["summary"]["num_of_reads"] = results[file_key]["num_of_reads"]
 
 
-def datadump(sampleComponentObj, log):
-    sampleComponentObj.start_data_dump(log=log)
-    sampleComponentObj.run_data_dump_on_function(extract_has_min_num_of_reads, log=log)
-    sampleComponentObj.end_data_dump(log=log)
-
+def datadump(samplecomponent_ref_json: Dict):
```

Comment:
I see datadump became longer again. If the code will be similar in other components it would make sense to move part of it to the library.